src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1334,7 +1334,11 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
                 DiagnosticLevel.Warning,
                 "BCP225",
                 $"The discriminator property \"{propertyName}\" value cannot be determined at compilation time. Type checking for this object is disabled.");
-
+            
+            public ErrorDiagnostic InvalidStringAddOperator() => new(
```

Comment:
>InvalidStringAddOperator [](http://example.com/codeflow?start=35&length=24)

The pattern we've been following for similar situation is to keep the first part of the original error message, but add an additional sentence to explain the special case.

In this case I'd change the message to this:
Cannot apply operator \"+" to operands of type \"{type1}\" and \"{type2}\". Use string interpolation instead.

This has the benefit of consistency with the other message.